todos/src/App.js

```diff
@@ -1,23 +1,45 @@
-import React from "react";
+import React, { useEffect, Fragment } from "react";
 import "App.css";
 import Navbar from "components/Navbar";
 import Todos from "components/Todos";
-import { Provider } from "react-redux";
-import store from "store";
 import TopContent from "components/TopContent";
 import { ToastContainer } from "react-toastify";
+import { useDispatch } from "react-redux";
+import { getCurrentTodos } from "supabaseData";
+import { addInitialData } from "actions/index";
+import { setSplashLoadingState } from "actions";
+import { useSelector } from "react-redux";
+import SplashScreen from "components/SplashScreen";
 
 function App() {
-  return (
-    <React.Fragment>
-      <Provider store={store}>
+  const splashLoadingState = useSelector(
+    (state) => state.laodingReducer.splashLoadingState
+  );
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const fetchData = async () => {
+      dispatch(setSplashLoadingState(true));
+      const initialData = await getCurrentTodos();
+      dispatch(setSplashLoadingState(false));
+      dispatch(addInitialData(initialData));
+    };
+    fetchData();
+  }, [dispatch]);
```

Comment:
Why are we watching the dispatch hook?

Reply:
I used that as suggested by my IDE. However, I realize that it is not necessary to include it in the dependency array since it does not change, and including it might cause unnecessary re-execution of the effect. Thank you for pointing it out. I will remove it from the dependency array.